CMakeLists.txt

```diff
@@ -66,6 +66,22 @@ option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
 
 find_package(CUDAToolkit REQUIRED)
 
+###################################################################################################
+# - find boost ------------------------------------------------------------------------------------
+
+find_package(Boost REQUIRED)
```

Comment:
Does this try and find _all_ of Boost? We only need the one header-only part for stacktrace.

Reply:
I'm not entirely sure. My understanding is that if I only want the headers, I don't call out COMPONENTS. I am just reading docs and stackoverflow though, so I'm far from an expert on cmake packages.

Reply:
This also doesn't fetch boost, it only finds it if it is installed...  Need to add it to third-party and use CPM I think.

Reply:
Yeah -- sadly, this is probably non-trivial. I believe that boost can not just be fetched and used via `add_subdirectory` like the other dependencies, so that'll require some more custom handling. I'll take a look.